components/console-backend-service/internal/domain/k8s/namespace_resolver_test.go

```diff
@@ -330,48 +310,79 @@ func TestNamespaceResolver_DeleteNamespace(t *testing.T) {
 		appRetriever := new(appAutomock.ApplicationRetriever)
 		svc.On("Find", name).Return(resource, nil).Once()
 		defer svc.AssertExpectations(t)
 
-		converter := automock.NewNamespaceConverter()
-		defer converter.AssertExpectations(t)
+		podSvc := automock.NewPodSvc()
+		resolver := k8s.NewNamespaceResolver(svc, appRetriever, []string{}, podSvc)
 
-		resolver := k8s.NewNamespaceResolver(svc, appRetriever)
-		converter.On("ToGQL", resource).Return(nil, errors.New("test error")).Once()
-		resolver.SetNamespaceConverter(converter)
+		svc.On("Delete", name).Return(errors.New("test error")).Once()
 
 		result, err := resolver.DeleteNamespace(nil, name)
 
 		require.Error(t, err)
 		assert.True(t, gqlerror.IsInternal(err))
 		assert.Nil(t, result)
 	})
+}
 
-	t.Run("Error deleting", func(t *testing.T) {
-		resource := fixNamespace(name, labels)
+func TestNamespaceResolver_NamespaceEventSubscription(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), -24*time.Hour)
+		cancel()
 
 		svc := automock.NewNamespaceSvc()
+		podSvc := automock.NewPodSvc()
+		svc.On("Subscribe", mock.Anything).Once()
+		svc.On("Unsubscribe", mock.Anything).Once()
+		podSvc.On("Subscribe", mock.Anything).Once()
+		podSvc.On("Unsubscribe", mock.Anything).Once()
+
 		appRetriever := new(appAutomock.ApplicationRetriever)
-		svc.On("Find", name).Return(resource, nil).Once()
-		defer svc.AssertExpectations(t)
+		resolver := k8s.NewNamespaceResolver(svc, appRetriever, []string{}, podSvc)
 
-		converter := automock.NewNamespaceConverter()
-		defer converter.AssertExpectations(t)
+		_, err := resolver.NamespaceEventSubscription(ctx, nil)
+		require.NoError(t, err)
 
-		resolver := k8s.NewNamespaceResolver(svc, appRetriever)
-		converter.On("ToGQL", resource).Return(&expected, nil).Once()
-		svc.On("Delete", name).Return(errors.New("test error")).Once()
-		resolver.SetNamespaceConverter(converter)
+		svc.AssertCalled(t, "Subscribe", mock.Anything)
+		podSvc.AssertCalled(t, "Subscribe", mock.Anything)
+	})
 
-		result, err := resolver.DeleteNamespace(nil, name)
+	t.Run("Unsubscribe after connection close", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), -24*time.Hour)
+		cancel()
 
-		require.Error(t, err)
-		assert.True(t, gqlerror.IsInternal(err))
-		assert.Nil(t, result)
+		svc := automock.NewNamespaceSvc()
+		podSvc := automock.NewPodSvc()
+		svc.On("Subscribe", mock.Anything).Once()
+		svc.On("Unsubscribe", mock.Anything).Once()
+		podSvc.On("Subscribe", mock.Anything).Once()
+		podSvc.On("Unsubscribe", mock.Anything).Once()
+
+		appRetriever := new(appAutomock.ApplicationRetriever)
+		resolver := k8s.NewNamespaceResolver(svc, appRetriever, []string{}, podSvc)
+
+		channel, err := resolver.NamespaceEventSubscription(ctx, nil)
+		require.NoError(t, err)
+
+		<-channel
+		svc.AssertCalled(t, "Unsubscribe", mock.Anything)
+		podSvc.AssertCalled(t, "Unsubscribe", mock.Anything)
```

Comment:
Also, the subscription runs in a different goroutine, and I worry that the code here may be subject to a race condition. Run `go test -count 100` to check if this test is stable.

Reply:
passed